File: nature-harvest-frontend/src/app/responses/user/login.response.ts
export class LoginResponse {
  message: string;
  token: string;
  refreshToken: string;
  tokenType: string;
  //user's detail
  id: string;
  username: string;
  role: string[];

  constructor(data: any) {
    this.message = data.message;
    this.token = data.token;
    this.refreshToken = data.refreshToken;
    this.tokenType = data.tokenType;
    this.id = data.id;
    this.username = data.username;
    this.role = data.role;
  }
}

File: nature-harvest-backend/nature-harvest-backend/src/main/java/com/api/nature_harvest_backend/responses/user/LoginResponse.java
package com.api.nature_harvest_backend.responses.user;
import lombok.*;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class LoginResponse {
        private String message;
        private String token;
        private String refreshToken;
        private String tokenType = "Bearer";
        //user's detail
        private String id;
        private String username;
        private List<String> role;
}
